presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -238,7 +240,7 @@ private static OrcPageSource createOrcPageSource(
             List<OrcColumn> fileReadColumns = new ArrayList<>(columns.size() + (isFullAcid ? 3 : 0));
             List<Type> fileReadTypes = new ArrayList<>(columns.size() + (isFullAcid ? 3 : 0));
             List<OrcReader.ProjectedLayout> fileReadLayouts = new ArrayList<>(columns.size() + (isFullAcid ? 3 : 0));
-            if (isFullAcid) {
+            if (isFullAcid && !originalFilesPresent) {
```

Comment:
we should verify expected schema also when handling original files

Reply:
fixed.